client/my-sites/themes/themes-magic-search-card/index.jsx

```diff
@@ -173,31 +157,31 @@ const ThemesMagicSearchCard = React.createClass( {
 				return <span className="themes-magic-search-card__search-text" key={ i }>{ token }</span>; // use shortid for key
 			} )
 		);
-	},
+	}
 
-	updateInput: function( updatedInput ) {
+	updateInput = ( updatedInput ) => {
 		this.setState( { searchInput: updatedInput } );
 		this.refs[ 'url-search' ].clear();
-	},
+	}
 
-	suggest: function( suggestion ) {
+	suggest = ( suggestion ) => {
 		const updatedInput = this.insertSuggestion( suggestion );
 		this.updateInput( updatedInput );
-	},
+	}
 
-	insertTextInInput: function( text ) {
+	insertTextInInput = ( text ) => {
 		const updatedInput = this.insertTextAtCursor( text );
 		this.updateInput( updatedInput );
-	},
+	}
 
 	render() {
 		const isJetpack = this.props.site && this.props.site.jetpack;
 		const isPremiumThemesEnabled = config.isEnabled( 'upgrades/premium-themes' );
 
 		const tiers = [
-			{ value: 'all', label: this.translate( 'All' ) },
-			{ value: 'free', label: this.translate( 'Free' ) },
-			{ value: 'premium', label: this.translate( 'Premium' ) },
+			{ value: 'all', label: this.props.translate( 'All' ) },
```

Comment:
Another small nit -- Might be a bit prettier if you define

```js
const translate = { this.props };
```

and use that here (and below, also for the `<Search>` `placeholder`)